src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.Digest.cs

```diff
@@ -227,17 +227,25 @@ private static string GetRandomAlphaNumericString()
 
         private static string ComputeHash(string data, string algorithm)
         {
-            // Disable MD5 insecure warning.
+            Span<byte> hashBuffer = stackalloc byte[SHA256.HashSizeInBytes]; // SHA256 is the largest hash produced
+            byte[] dataBytes = Encoding.UTF8.GetBytes(data);
```

Comment:
Can this allocation be avoided?
E.g. by using a buffer from the array-pool or stack allocating for smaller data-length.

Reply:
Probably. Honestly this whole file has a lot of opportunities to reduce string allocations. I held off on that since I was focusing just on how `HashAlgorithm` is being used throughout `libraries`.